docs/guides/getting-started/getting-started-scaling-and-tuning.adoc

```diff
@@ -0,0 +1,62 @@
+
+While there is some automatic adjustment based upon how many cores you make available to your {project_name} server, fine-tuning thread counts can improve overall throughput see <@links.ha id="concepts-threads" /> - but this must be done in conjunction with other VM resources, such as database connections - see <@links.ha id="concepts-database-connections" /> otherwise you may be moving a bottleneck somewhere else.
+
+To limit memory utilization of queued work and to provide for load shedding see <@links.ha id="concepts-threads#load-shedding" /> 
+
+If you are experiencing timeouts in obtaining database connections, you should consider increasing the number of connections available - see <@links.ha id="concepts-database-connections" />.
+
+=== Vertical Autoscaling
+
+Some platforms, such as Kubernetes, provide mechanisms to vertically autoscale. That is not recommended for {project_name} if it requires restarting the server instance - which is currently the case for Java on Kubernetes. You can consider instead providing higher CPU and/or memory limits to your container to allow your JVM to adapt within those limits as needed.
+
+== Horizontal Scaling
+
+A single {project_name} instance is susceptible to availability issues. If the instance goes down you will be faced with a full outage, and until the persistent sessions feature is available all of your on-going sessions will be lost. By running at least two cluster members on different machines you will greatly increase the availability of {project_name}. 
+
+There will also be a limit on the amount of concurrent requests a single VM can effectively handle. Additional server instances can provide roughly linear scaling of throughput until associated resources, such as the database or distributed caching limit scaling.
+
+In general it's best to let the {project_name} Operator handle horizontal scaling concerns. When using the Operator simply set the Keycloak custom resource spec.instances as desired to horizontally scale - see <@links.ha id="deploy-keycloak-kubernetes" /> for more.
+
+If you are not using the Operator, please review the following.
+
+Your instances should be on separate machines if possible to increase availability. On Kubernetes you can use Pod anti-affinitity to enforce this.
+
+You need to use distributed or external caching for cluster members to share the same session state - see <@links.server id="caching" /> for more on the relevant configuration. The embedded Infinispan cache has horizontal scaling considerations including:
+- Your instances will need a way to find each other, see discovery in the server guide.
+- It is not well suited for clusters crossing availability zones, also known as a stretch clusters. It is recommended that when using the embedded Infinispan cache that the instances all exist in the same availability zone. On Kubernetes you can use pod affinity to enforce this grouping of pods.
```

Comment:
`/clusters crossing availability zones/clusters spanning multiple availability zones`?

Reply:
Added the suggestion.